### ISSUE TITLE
Quantum Tunnels for Mobile 

### ISSUE THREAD

**Issue by TempUser13** [mullvad/mullvadvpn-app#4932]
Are there plans or things in the works for Quantum Tunnels on mobile? Are there any limitations or is this just one of the things on the to do list? Thanks for all the pre-existing hard work and service.

**Comment by faern** [comment]
Quantum resistant tunneling support for Android has [already been merged to `main`](https://github.com/mullvad/mullvadvpn-app/pull/4742). However, it is not yet released. An upcoming release I can't give you the date for will include it. It will eventually come to iOS as well, but it is not scheduled for Q3 at least.

**Comment by TempUser13** [comment]
Thank you for the reply, is there an android canary build available to be able to use quantum tunnels until the feature is officially released in stable?

**Comment by faern** [comment]
There are no betas available yet or any builds we have run though our pre-release QA and testing. However, all our latest code is always built and uploaded to https://releases.mullvad.net/builds/. However, we do not recommend running that unless you know what you are doing and are OK with bugs. We do not have any security or stability guarantees for the builds available there.

I would recommend that you wait until we have at least a beta with this feature in it. I can't imagine you would need PQ crypto so much that you are willing to risk introducing other potentially security related bugs into your device.

**Comment by albin-mullvad** [comment]
A beta with this feature is now available on GitHub: https://github.com/mullvad/mullvadvpn-app/releases/tag/android%2F2023.6-beta1

**Comment by thekeymaker33** [comment]
When does the Quantum Tunnels come to iOS?

**Comment by faern** [comment]
It's scheduled to be worked on in Q1 of 2024. So **hopefully** first half of next year.

**Comment by PaulCharlton** [comment]
has the IOS worked commenced yet?

**Comment by PaulCharlton** [comment]
has the IOS work commenced yet? I have some bandwidth and may be able to implement the IOS version.

**Comment by buggmagnet** [comment]
Hello @PaulCharlton, @thekeymaker33  and everyone else.
As @faern wrote back in December, we are now working on enabling Quantum-resistant tunnelling on iOS.
Whilst I can't comment on a timeline, it should be coming soon.

@PaulCharlton we really appreciate your enthusiasm for working on it, but we decided that it will be faster and easier for us if we work on this internally. I want to emphasise that we are not refusing your help, but just that the timing is unfortunate.

**Comment by PaulCharlton** [comment]
@buggmagnet you may want to review #5801 anyway ... most UI changes already made and merged with the latest main which picked up some of your changes.  conflicts resolved.

Naming of variables is an issue since most existing settings for quantum are scoped under "wireGuard" and the other settings names are therefore prefixed with "wireGuard" so I chose to prefix the quantum tunnel objects/names with "wireGuard" and keep the suffices consistent with "xxxSettings", etc.

**Comment by PaulCharlton** [comment]
@buggmagnet no big deal on the duplication of effort -- so far I have spent less than a day total, most of which was grokking the high level design from the code project. re: #5801

**Comment by agnosticlines** [comment]
Hey, is there any update on this? Love the work you're all doing and would love to use the PQ tunnels on my phone

**Comment by thekeymaker33** [comment]
> Hey, is there any update on this? Love the work you're all doing and would love to use the PQ tunnels on my iPhone

The beta version has been released, you can install and test it via TestFlight.

I think the full version should be released soon.

**Comment by agnosticlines** [comment]
> > Hey, is there any update on this? Love the work you're all doing and would love to use the PQ tunnels on my iPhone
> 
> 
> 
> The beta version has been released, you can install and test it via TestFlight.
> 
> 
> 
> I think the full version should be released soon.

Oh great! Thanks for sharing, unfortunately the beta is full but excited to see this fully release :-)

**Comment by buggmagnet** [comment]
Hello @thekeymaker33 @agnosticlines and everyone else. 
As we wrote in [our blog post](https://mullvad.net/en/blog/leaks-in-ios-beta-release-testflight-20244-1) 
We found an issue in our Test Flight build where we were leaking a bit of traffic when entering a post quantum secure connection state.
We consider this a serious issue, and we wanted to make sure this is fixed before releasing anything.
Alongside this issue, we found another issue where a timed out connection would not recover properly, which we just fixed. 

Whilst I cannot confirm any date, I can say that we will be looking at creating another TestFlight build soon, and then make a release out of it, if we do not see more problems arising. 

Thank you for the support, we appreciate it!
